mlir/lib/Dialect/MIOpen/Transforms/AlignTiling.cpp

```diff
@@ -469,9 +476,109 @@ LogicalResult MILARewritePattern::matchAndRewrite(linalg::GenericOp laGeneric,
   return failure();
 }
 
+void eliminateAll(Operation *op) {
+  for (Operation *user : op->getUsers()) {
+    eliminateAll(user);
+  }
+  op->dropAllUses();
+  op->erase();
+}
+
+// Dropping expanded dims from linalg elementwise fusion
+LogicalResult LGDropDimsPattern::matchAndRewrite(linalg::GenericOp laGeneric,
```

Comment:
Please explain why this can't be done during the main fusion process. Furthermore, please explain why this sort of expansive rewriting of the generic, instead of a more targeted transformation, is required.

Reply:
It might be possible to include this into the main fusion align but I'm trying not to.
The reason is, I like the way you changed the fusion recently and want to keep the main fusion as simple as possible.

Linalg already has very similar pass but it doesn't perfectly fit to my purpose.
https://mlir.llvm.org/docs/Passes/#-linalg-fold-unit-extent-dims-remove-unit-extent-dimension-in-linalg-ops-on-tensors

I don't think this is a hacky solution but a normal use case of the multi-level compilation to tame the IR from different level of compilation.
Maybe there is a better way to apply this transform as you mentioned above, please let me know.